File: 0/Templates/VisualBasic/ProjectTemplates/CompositionAppVBTemplate/MainView.vb
Imports System.Numerics
Imports Windows.ApplicationModel.Core
Imports Windows.UI
Imports Windows.UI.Composition
Imports Windows.UI.Core

Public Class MainView
    Implements IFrameworkView

    Private view As CoreApplicationView
    Private window As CoreWindow
    Private compositor As Compositor
    Private compositionTarget As CompositionTarget
    Private root As ContainerVisual
    Private background As SpriteVisual
    Private target As SpriteVisual

    Private Sub InitializeComposition()
        ' setup compositor And root visual
        compositor = New Compositor()
        root = compositor.CreateContainerVisual()

        ' associate with the CoreWindow
        compositionTarget = compositor.CreateTargetForCurrentView()
        compositionTarget.Root = root

        ' add a solid color background
        background = compositor.CreateSpriteVisual()
        background.Brush = compositor.CreateColorBrush(Colors.LightGreen)
        root.Children.InsertAtBottom(background)

        ' create green square
        target = compositor.CreateSpriteVisual()
        target.Brush = compositor.CreateColorBrush(Colors.Green)
        target.Size = New Vector2(150.0F, 150.0F)
        target.Offset = New Vector3(250.0F, 250.0F, 0.0F)
        target.CenterPoint = New Vector3(75.0F, 75.0F, 0.0F)
        root.Children.InsertAtTop(target)

        ' animate square
        Animate(target)

        UpdateSize()
    End Sub

    Private Sub Animate(visual As Visual)
        Dim easing = compositor.CreateCubicBezierEasingFunction(New Vector2(0.5F, 0.1F), New Vector2(0.5F, 0.75F))
        Dim animation = compositor.CreateScalarKeyFrameAnimation()

        animation.InsertKeyFrame(0.00F, 0.00F, easing)
        animation.InsertKeyFrame(1.0F, 360.0F, easing)

        animation.Duration = TimeSpan.FromMilliseconds(2000)
        animation.IterationBehavior = AnimationIterationBehavior.Forever

        visual.StartAnimation("RotationAngleinDegrees", animation)
    End Sub

    Private Sub UpdateSize()
        If (background Is Nothing Or window Is Nothing) Then
            Return
        End If

        background.Size = New Vector2(window.Bounds.Width, window.Bounds.Height)
    End Sub

    Private Sub OnWindowSizeChanged(sender As CoreWindow, args As WindowSizeChangedEventArgs)
        UpdateSize()
    End Sub

    Public Sub Initialize(applicationView As CoreApplicationView) Implements IFrameworkView.Initialize
        view = applicationView
    End Sub

    Public Sub Load(entryPoint As String) Implements IFrameworkView.Load
        ' not used
    End Sub

    Public Sub Run() Implements IFrameworkView.Run
        window.Activate()
        window.Dispatcher.ProcessEvents(CoreProcessEventsOption.ProcessUntilQuit)
    End Sub

    Public Sub SetWindow(window As CoreWindow) Implements IFrameworkView.SetWindow
        Me.window = window
        AddHandler window.SizeChanged, AddressOf OnWindowSizeChanged

        InitializeComposition()
    End Sub

    Public Sub Uninitialize() Implements IFrameworkView.Uninitialize
        window = Nothing
        view = Nothing
    End Sub
End Class

File: 0/Templates/CSharp/ProjectTemplates/CompositionAppTemplate/MainView.cs
using System;
using System.Numerics;
using Windows.ApplicationModel.Core;
using Windows.UI;
using Windows.UI.Composition;
using Windows.UI.Core;

namespace $safeprojectname$
{
    class MainView : IFrameworkView
    {
        private CoreApplicationView view;
        private CoreWindow window;
        private Compositor compositor;
        private CompositionTarget compositionTarget;
        private ContainerVisual root;
        private SpriteVisual background;
        private SpriteVisual target;

        private void InitializeComposition()
        {
            // setup compositor and root visual
            this.compositor = new Compositor();
            this.root = this.compositor.CreateContainerVisual();

            // associate with the CoreWindow
            this.compositionTarget = this.compositor.CreateTargetForCurrentView();
            this.compositionTarget.Root = this.root;

            // add a solid color background
            this.background = this.compositor.CreateSpriteVisual();
            this.background.Brush = this.compositor.CreateColorBrush(Colors.LightGreen);
            this.root.Children.InsertAtBottom(this.background);

            // create green square
            var colorVisual = this.compositor.CreateSpriteVisual();
            colorVisual.Brush = this.compositor.CreateColorBrush(Colors.Green);
            colorVisual.Size = new Vector2(150.0f, 150.0f);
            colorVisual.Offset = new Vector3(250.0f, 250.0f, 0.0f);
            colorVisual.CenterPoint = new Vector3(75.0f, 75.0f, 0.0f);
            this.target = colorVisual;
            this.root.Children.InsertAtTop(this.target);

            // animate square
            Animate(this.target);

            UpdateSize();
        }

        private void Animate(Visual visual)
        {
            var easing = this.compositor.CreateCubicBezierEasingFunction(new Vector2(.5f, .1f), new Vector2(.5f, .75f));
            var animation = this.compositor.CreateScalarKeyFrameAnimation();

            animation.InsertKeyFrame(0.00f, 0.00f, easing);
            animation.InsertKeyFrame(1.00f, 360.0f, easing);

            animation.Duration = TimeSpan.FromMilliseconds(2000);
            animation.IterationBehavior = AnimationIterationBehavior.Forever;

            visual.StartAnimation("RotationAngleinDegrees", animation);
        }

        private void UpdateSize()
        {
            if (this.background != null && this.window != null) {
                this.background.Size = new Vector2((float)this.window.Bounds.Width, (float)this.window.Bounds.Height);
            }
        }

        private void OnWindowSizeChanged(CoreWindow sender, WindowSizeChangedEventArgs args)
        {
            UpdateSize();
        }

        /// <summary>
        /// A method for app view initialization, which is called when an app object is launched.
        /// </summary>
        /// <param name="view">The default view provided by the app object.</param>
        void IFrameworkView.Initialize(CoreApplicationView view)
        {
            this.view = view;
        }

        /// <summary>
        /// Loads or activates any external resources used by the app view before Run is called.
        /// </summary>
        /// <param name="entryPoint">The name of the entry point method for the activated type.</param>
        void IFrameworkView.Load(string entryPoint)
        {
            // not used
        }

        /// <summary>
        /// Starts the app view.
        /// </summary>
        void IFrameworkView.Run()
        {
            this.window.Activate();
            this.window.Dispatcher.ProcessEvents(CoreProcessEventsOption.ProcessUntilQuit);
        }

        /// <summary>
        /// Sets the current window for the app object's view.
        /// </summary>
        /// <param name="window">The current window for the app object.</param>
        void IFrameworkView.SetWindow(CoreWindow window)
        {
            this.window = window;
            this.window.SizeChanged += OnWindowSizeChanged;
            InitializeComposition();
        }

        /// <summary>
        /// Uninitializes the app view.
        /// </summary>
        void IFrameworkView.Uninitialize()
        {
            this.window = null;
            this.view = null;
        }
    }
}
